tools/clang/lib/AST/HlslTypes.cpp

```diff
@@ -704,32 +664,27 @@ bool IsHLSLObjectWithImplicitROMemberAccess(clang::QualType type) {
 }
 
 bool IsHLSLRWNodeInputRecordType(clang::QualType type) {
-  if (const RecordType *RT = type->getAs<RecordType>()) {
-    StringRef name = RT->getDecl()->getName();
-    if (name == "RWDispatchNodeInputRecord" ||
-        name == "RWGroupNodeInputRecords" || name == "RWThreadNodeInputRecord")
-      return true;
-  }
+  if (const HLSLNodeObjectAttr *Attr = getNodeAttr(type))
+    return Attr->getType() == HLSLNodeObjectAttr::RWDispatchNodeInputRecord ||
+           Attr->getType() == HLSLNodeObjectAttr::RWGroupNodeInputRecords ||
+           Attr->getType() == HLSLNodeObjectAttr::RWThreadNodeInputRecord;
   return false;
 }
 
 bool IsHLSLRONodeInputRecordType(clang::QualType type) {
-  if (const RecordType *RT = type->getAs<RecordType>()) {
-    StringRef name = RT->getDecl()->getName();
-    if (name == "DispatchNodeInputRecord" || name == "GroupNodeInputRecords" ||
-        name == "ThreadNodeInputRecord")
-      return true;
-  }
+  if (const HLSLNodeObjectAttr *Attr = getNodeAttr(type))
+    return Attr->getType() == HLSLNodeObjectAttr::DispatchNodeInputRecord ||
+           Attr->getType() == HLSLNodeObjectAttr::GroupNodeInputRecords ||
+           Attr->getType() == HLSLNodeObjectAttr::ThreadNodeInputRecord;
```

Comment:
As above, to save you the trouble: 

```c++

```c++
return (static_cast<uint32_t>(GetNodeIOType(type)) &
          (static_cast<uint32_t>(DXIL::NodeIOFlags::ReadWrite) |  static_cast<uint32_t>(DXIL::NodeIOFlags::Input))) ==
           static_cast<uint32_t>(DXIL::NodeIOFlags::Input);
```